apps/speed.c

```diff
@@ -175,59 +175,73 @@ typedef struct loopargs_st {
     GCM128_CONTEXT *gcm_ctx;
 } loopargs_t;
 
+typedef int (*loop_proc_fn)(int testnum, loopargs_t *args);
+
+typedef struct loop_ctx_st {
+    int testnum;
```

Comment:
@richsalz I can remove the `testnum` commit to keep this PR simple, if you wish ?